msteams-platform/bots/streaming-ux.md

```diff
@@ -0,0 +1,586 @@
+ms.topic: conceptual
+author: surbhigupta12
+ms.author: surbhigupta
+ms.localizationpriority: high
+---
+
+# Stream bot messages
+
+>[!NOTE]
+>
+> - Streaming bot messages is available only for one-on-one chats and in [public developer preview](../resources/dev-preview/developer-preview-intro.md).
+> - Streaming bot messages isn't available with Tools and the `o1` model.
+
+You can stream bot messages to deliver a bot's responses to the user as small updates while the complete response is being generated to enhance the user experience. Often, bots take a long time to generate responses without updating the user interface, leading to a less engaging experience.
+
+When users observe the bot processing their request in real time, it can increase their satisfaction and trust. This perceived responsiveness and transparency enhances user engagement and decreases conversation abandonment with the bot.
+
+Streaming bot messages has two types of updates:
+
+- **Informative updates**: Informative updates appear as a blue progress bar at the bottom of the chat. It informs the user about the bot's ongoing actions while a response is being generated.
+
+  :::image type="content" source="../assets/images/bots/stream_type_informative.png" alt-text="Screenshot shows the bots informative updates of streaming." lightbox="../assets/images/bots/stream_type_informative.png":::
+
+- **Response streaming**: Response streaming is displayed as a typing indicator. It reveals the bot's response to the user as small updates while the complete response is being generated.
+
+  :::image type="content" source="../assets/images/bots/stream_type_responsive.png" alt-text="Screenshot shows the bots response streaming." lightbox="../assets/images/bots/stream_type_responsive.png":::
+
+You can implement streaming bot messages in your app in one of the following ways:
+
+- [Stream using Teams AI library](#stream-message-through-teams-ai-library).
+- [Stream using REST API](#stream-message-through-rest-api).
+
+## Stream message through Teams AI library
+
+The Teams AI library provides the capability to stream messages for AI-powered bots. Streaming bot messages helps to ease the response time lag while the Large Language Model (LLM) generates the complete response. The primary factors contributing to slow response time include multiple preprocessing steps, such as Retrieval-Augmented Generation (RAG) or function calls, and the time required by the LLM to generate a full response.
+
+Through streaming, your AI bot can offer an experience that is engaging and responsive for the user. To configure streaming messages for your AI-powered bot app:
+
+1. **Enable streaming for AI bot**:
+
+    Bot messages can be streamed through AI SDK. The AI-powered bot sends chunks to the user as the model generates the response. Streaming messages support rich text and citation. However, attachment, AI-label, feedback loop, and sensitivity labels are available only for the final streaming message.
```

Comment:
Update to 

`Bot messages can be streamed through AI SDK. The AI-powered bot sends chunks to the user as the model generates the response. Streaming messages support text. However, attachments, AI-label, feedback loop, citations, and sensitivity labels are available only for the final streaming message.`

Reply:
@lilyydu As per the loop, streaming messages support rich HTML and Citations. Hence, I added it here. Please confirm if this not supported.

Reply:
Updated as discussed.